hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go

```diff
@@ -156,15 +172,32 @@ func (r *DedicatedServingComponentScheduler) Reconcile(ctx context.Context, req
 				// No zone has been set on the node, we cannot use it
 				continue
 			}
+
 			_, hasHCLabel := node.Labels[hyperv1.HostedClusterLabel]
 			if hasHCLabel {
 				// The node has been allocated to a different hosted cluster, skip it
 				continue
 			}
+
 			if nodesToUse[zone] == nil {
+
+				// if the candidate Node is not paired with the existing node to use then skip.
+				paired := false
+				if len(nodesToUse) > 0 {
+					for _, n := range nodesToUse {
+						if n.Labels[OSDFleetManagerPairedNodesLabel] == node.Labels[OSDFleetManagerPairedNodesLabel] {
```

Comment:
the list can only have 1 entry at this point, can't we directly check the first node?
```go
if nodesToUse[0].Labels[OSDFleetManagerPairedNodesLabel] == node.Labels[OSDFleetManagerPairedNodesLabel] ...
```

lgtm otherwise

Reply:
it's a map, not a slice so that's not possible. But I'm happy to change if if there's a better way.

Reply:
oh right, I missed that its a map :) 

One more question, with this we can only pickup a pair of nodes which is dictated by `OSDFleetManagerPairedNodesLabel` label, so do we still need to check that the nodes are in different zones?

Reply:
good point, we could let the paired label to be the only source of truth and let the distribution of it as an exercise for the owner. In practice it shouldn't make a difference. It doesn't hurt keeping it though. @mjlshen @minonne thoughts?

Reply:
The nodes definitely should be in different zones if they match up this way, but since we don't automatically remediate situations where podAntiAffinity isn't satisfied because they're in the same zone, I think it's better to leave it for now as an additional safeguard

Reply:
I agree with @mjlshen . If everything goes right none will spot any differences but in case anything goes wrong (same label in the same zone) the POD will stay `Pending` right?
I would keep it like this